### PR TITLE
initial changes for neuraldsde and neuralsde

### PR DESCRIPTION
**Pull request by Abhishek-1Bhatt**

### CODE REVIEW

File: src/neural_de.jl [hunk 1]
```diff
@@ -179,19 +180,31 @@ function (n::NeuralDSDE{M})(x,p=n.p) where {M<:FastChain}
...
-      u_, st2 = n.model2(u,p[2],st2)
+    function g(u,p,t;st=st)
+      u_, st.state2 = n.model2(u,p.p2,st.state2)
       return u_
     end
     
     ff = SDEFunction{false}(dudt_,g,tgrad=basic_tgrad)
     prob = SDEProblem{false}(ff,g,x,n.tspan,p)
-    return solve(prob,n.args...;sensealg=TrackerAdjoint(),n.kwargs...), st1, st2
+    return solve(prob,n.args...;sensealg=TrackerAdjoint(),n.kwargs...), st
```
Comment (Abhishek-1Bhatt) [inline src/neural_de.jl:207]: I've made some changes here in accordance with Lux api requirement, currently only for NeuralDSDE and NeuralSDE layers. However there is something not done quite correctly here as calling Lux.setup on NeuralDSDE still return `(NamedTuple(), NamedTuple())`  cc @ChrisRackauckas

File: src/neural_de.jl [hunk 2]
```diff
@@ -179,19 +180,31 @@ function (n::NeuralDSDE{M})(x,p=n.p) where {M<:FastChain}
...
-function (n::NeuralDSDE{M})(x,p,st1,st2) where {M<:Lux.AbstractExplicitLayer}
-    function dudt_(u,p,t)
-      u_, st1 = n.model1(u,p[1],st1)
+function initialparameters(rng::AbstractRNG, n::NeuralDSDE)
+    p1 = Lux.initialparameters(rng, n.model1)
+    p2 = Lux.initialparameters(rng, n.model2)
+    return Lux.ComponentArray((p1 = p1, p2 = p2))
+end
+
+function initialstates(rng::AbstractRNG, n::NeuralDSDE)
```
Comment (ChrisRackauckas) [inline src/neural_de.jl:189]:
```suggestion
function Lux.initialparameters(rng::AbstractRNG, n::NeuralDSDE)
    p1 = Lux.initialparameters(rng, n.model1)
    p2 = Lux.initialparameters(rng, n.model2)
    return Lux.ComponentArray((p1 = p1, p2 = p2))
end

function Lux.initialstates(rng::AbstractRNG, n::NeuralDSDE)
```

File: src/neural_de.jl [hunk 3]
```diff
@@ -269,6 +291,32 @@ function (n::NeuralSDE{P,M})(x,p=n.p) where {P,M<:FastChain}
     solve(prob,n.args...;sensealg=TrackerAdjoint(),n.kwargs...)
 end
 
+function initialparameters(rng::AbstractRNG, n::NeuralSDE)
+    p1 = initialparameters(rng, n.model1)
+    p2 = initialparameters(rng, n.model2)
+    return Lux.ComponentArray((p1 = p1, p2 = p2))
+end
+function initialstates(rng::AbstractRNG, n::NeuralSDE)
```
Comment (ChrisRackauckas) [inline src/neural_de.jl:299]: shadowed not extended here too

File: docs/src/examples/neural_sde.md [hunk 4]
```diff
@@ -76,28 +76,24 @@ neural SDE with diagonal noise layer function:
...
 
 diffusion_dudt = Lux.Chain(Lux.Dense(2, 2))
-p2, st2 = Lux.setup(rng, diffusion_dudt)
 
-p1 = Lux.ComponentArray(p1)
-p2 = Lux.ComponentArray(p2)
-#Component Arrays doesn't provide a name to the first ComponentVector, only subsequent ones get a name for dereferencing
-p = [p1, p2]
 
 neuralsde = NeuralDSDE(drift_dudt, diffusion_dudt, tspan, SOSRI(),
```
Comment (Abhishek-1Bhatt) [inline docs/src/examples/neural_sde.md:83]: The choice of the solver here is leading to https://github.com/SciML/DiffEqFlux.jl/runs/7005695604?check_suite_focus=true#step:5:22
We'll either need a different sensealg for NeuralDSDE(currently InterpolatingAdjoint) which is compatible with Lux or change the solver

**Pull request closed without merging**